packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -476,6 +483,67 @@ - (FlutterError *_Nullable)onListenWithArguments:(id _Nullable)arguments
   return nil;
 }
 
+
+- (NSArray<FLTGetEmbeddedSubtitlesMessage *> *) getEmbeddedSubtitles {
+    NSArray<AVMediaCharacteristic> *characteristics= _player.currentItem.asset.availableMediaCharacteristicsWithMediaSelectionOptions;
+    
+    
+    for(int i=0;i<[characteristics count];i++){
+        AVMediaCharacteristic characteristic = characteristics[i];
```

Comment:
you can use `enumerateObjectsUsingBlock` to access both index and the element itself